Added the --no-headless option back in (#3112)
elastic/observability-docs: docs/en/observability/synthetics-command-reference.asciidoc

<<synthetics-configuration-monitor,`monitor.throttling`>>.
The value defined via the CLI will take precedence.

`--no-headless`::
Runs with the browser in headful mode.
+

> This is the same as running `--playwright-options '{"headless": false}'`.

Reviewer: Its good to inform users that we don't support running headful on the Synthetics global infrastructure. 

This is one of the main reason why I removed this flag as it was not uniform.

Reviewer: My worry is that, people start using it and not seeing results in their monitor. Not worried about CLI/PW Options, but just as a Note saying we dont support headful mode in our infrastructure.

Author: @vigneshshanmugam good point, I've added a note to that effect if you could do another review

Reviewer 2: Do you think the audience for this doc will already be familiar with the terms "headless" and "headful"? If not, a link to the Playwright docs might be helpful. I'll leave it up to you!

```suggestion
This is the same as setting https://playwright.dev/docs/api/class-testoptions#test-options-headless[Playwright's `headless` option] to `false` by running `--playwright-options '{"headless": false}'`.
```

<<synthetics-configuration-monitor,`monitor.throttling`>>.
The value defined via the CLI will take precedence.

`--no-headless`::
Runs with the browser in headful mode.
+
This is the same as running `--playwright-options '{"headless": false}'`.

[NOTE]
=====

> Headful mode is only to be used locally when testing, to help see the browser being shown and to interact with DOM elements. Do not attempt to run in headful mode when running through Elastic's global managed testing infrastructure, or {private-location}s.

Reviewer 2: ```suggestion
Headful mode should only be used locally to see the browser and interact with DOM elements directly for testing purposes. Do not attempt to run in headful mode when running through Elastic's global managed testing infrastructure or {private-location}s.
```

Reviewer: Can we also mention that we dont support on our infrastructure instead of saying not to use it.
